File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int left = 0;
        int right = 1;
        while(right < prices.size())
        {
            int currentProfit = prices[right] - prices[left];
            if (currentProfit > profit) profit = currentProfit;
            if (prices[left] > prices[right])
            {
                left = right;
                right += 1;
            }
            else
                right++;
        }
        return (profit <= 0 ? 0 : profit);
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        # for this one we must use BFS with queue and also group nodes at each level
        if not root:
            return []
        res = []
        q = deque([root])
        while q:
            level = len(q)
            level_nodes = []
            for _ in range(level):
                node = q.popleft()
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
                level_nodes.append(node.val)
            res.append(level_nodes)
        return res